src/me/mrCookieSlime/Slimefun/SlimefunGuide.java

```diff
@@ -231,60 +231,69 @@ public static void openCredits(Player p, final ItemStack guide) {
 				}
 				return false;
 			});
-			
+
 			index++;
 		}
-		
+
 		menu.open(p);
 	}
 
 	public static void openCheatMenu(Player p) {
-		openMainMenu(p, false, false, 1);
+		openMainMenuAsync(p, false, false, 1);
 	}
-	
+
 	public static void openGuide(Player p, boolean book) {
 		if (!SlimefunPlugin.getWhitelist().getBoolean(p.getWorld().getName() + ".enabled")) return;
 		if (!SlimefunPlugin.getWhitelist().getBoolean(p.getWorld().getName() + ".enabled-items.SLIMEFUN_GUIDE")) return;
-		
-		if (!getHistory().containsKey(p.getUniqueId())) {
-			openMainMenu(p, true, book, 1);
-		}
-		else {
-			Object last = getLastEntry(p, false);
+
+		Object last = getLastEntry(p, false);
+
+		if (last == null)
+			openMainMenuAsync(p, true, book, 1);
+		else
 			handleHistory(p, last, book, false);
-		}
 	}
 
-	public static void openMainMenu(final Player p, final boolean survival, final boolean book, final int selected_page) {
+	private static void openMainMenuAsync(final Player player, final boolean survival, final boolean book, final int selected_page) {
+		Slimefun.getLocal().sendMessage(player, "messages.opening-guide");
```

Comment:
I think this message should only be displayed when the Person is opening the guide for the first time.
Otherwise there won't be any delay anyway.

Reply:
Was thinking of the best way to do this, got a suggestion?

Reply:
BiConsumer with a Boolean? XD

Reply:
Also thought about it but... it'd be useless for 90% of cases so I'd need another method